src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,51 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void validateUser(User user) {
+        if (user == null) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "User must not be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "Login should not be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "Password should not be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "Age should not be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "User with login " + user.getLogin() + " already exists.");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "Login should have at least 6 characters.");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "Password should have at least 6 characters.");
```

Comment:
The checklist mentions testing edge cases for password length. The condition here is correct, but make sure that the corresponding tests for edge cases (password lengths of 0, 5, 6, and more than 6) are included in the test suite.